app/plugins/eu.esdihumboldt.hale.app.transform/src/eu/esdihumboldt/hale/app/transform/ExecTransformation.java

```diff
@@ -457,19 +438,89 @@ private void transform() throws InterruptedException, ExecutionException {
 		ListenableFuture<Boolean> res = Transformation.transform(sources, target, env,
 				reportHandler, id, validators, context.getFilters());
 
-		if (res.get()) {
+		boolean orgSuccess = res.get();
+
+		// Job threads might still be active, wait a moment to allow
+		// them to complete and file their report (otherwise error may get lost)
+		try {
+			Thread.sleep(3000);
+		} catch (Throwable e) {
+			// ignore
+		}
+
+		boolean success;
+		try {
+			success = evaluateSuccess(orgSuccess);
+		} catch (Throwable e) {
+			error("Success evaluation resulted in an error:\n" + e.getMessage());
+			if (context.isLogException()) {
+				e.printStackTrace();
+			}
+			return 2;
+		}
+
+		if (success) {
 			info("Transformation completed. Please check the reports for more details.");
 		}
 		else {
-			// Job threads might still be active, wait a moment to allow them to
-			// complete and file their report (otherwise error may get lost)
-			try {
-				Thread.sleep(3000);
-			} catch (Throwable e) {
-				// ignore
+			if (orgSuccess) {
+				error("Transformation failed according to the success evaluation script.");
+				return 2;
+			}
+			else {
+				error("Transformation failed, please check the reports for details.");
+				return 1;
+			}
+		}
+
+		// exit OK
+		return 0;
+	}
+
+	private boolean evaluateSuccess(boolean success) {
+		StatsCollector statistics = reportHandler.getStatistics();
+
+		if (context.getStatisticsFile() != null) {
+			// write to statistics file
+			try (Writer writer = Files.newBufferedWriter(context.getStatisticsFile().toPath(),
+					StandardCharsets.UTF_8)) {
+				writer.write(JsonOutput.prettyPrint(statistics.saveToJson(true)));
+			} catch (IOException e) {
+				ExecUtil.error("Error writing statistics file: " + e.getMessage());
+				// XXX should this fail the command?
 			}
-			throw fail("Transformation failed, please check the reports for details.");
 		}
+
+		if (context.getSuccessEvaluationScript() == null) {
+			// no custom evaluation script
+			return success;
+		}
+
+		// run evaluation script
+
+		CompilerConfiguration compilerConfiguration = new CompilerConfiguration();
+		compilerConfiguration.setScriptBaseClass(DelegatingScript.class.getName());
+
+		// Configure the GroovyShell and pass the compiler configuration.
+		GroovyShell shell = new GroovyShell(getClass().getClassLoader(), new Binding(),
+				compilerConfiguration);
+		DelegatingScript script;
+		try (InputStream in = new DefaultInputSupplier(context.getSuccessEvaluationScript())
+				.getInput();
+				InputStreamReader reader = new InputStreamReader(in, StandardCharsets.UTF_8)) {
```

Comment:
Should we make the charset of the evaluation script configurable? Since it's a user input it can't be guaranteed to be in UTF-8.

Reply:
I think it is OK if it is not configurable. UTF-8 is often compatible with the default charsets and usage of special characters is unlikely for the logic (though they may be in comments).